ppl/src/main/java/org/opensearch/sql/ppl/parser/AstBuilder.java

```diff
@@ -307,14 +308,30 @@ protected UnresolvedPlan aggregateResult(UnresolvedPlan aggregate, UnresolvedPla
     return aggregate;
   }
 
+  /**
+   * Kmeans command.
+   */
   @Override
   public UnresolvedPlan visitKmeansCommand(KmeansCommandContext ctx) {
-    return new Kmeans(ArgumentFactory.getArgumentList(ctx));
+    return new Kmeans(ctx.kmeansParameter().stream()
```

Comment:
is it possible that ctx.kmeansParameter() == null, for example, `kmeans` without parameters.

Reply:
if ctx.kmeansParameter() == null, ctx.kmeansParameter().stream() will throw null pointer exception?

Reply:
I think ctx.kmeansParameter() is empty rather than null when `kmeans` has no parameter. Will verify with debugger.